instrumentation/excon/lib/opentelemetry/instrumentation/excon/middlewares/tracer_middleware.rb

```diff
@@ -71,43 +77,35 @@ def self.around_default_stack
           private
 
           def handle_response(datum)
-            if datum.key?(:otel_span)
-              datum[:otel_span].tap do |span|
-                return span if span.end_timestamp
+            datum.delete(:otel_span)&.tap do |span|
+              return span unless span.recording?
```

Comment:
I was a little confused about it returning the span explicitly. 

This is really intended to be a short cut to exit the `handle_response` method correct?

```suggestion
              return unless span.recording?
```

Reply:
I believe this returned the span because [Span#finish](https://www.rubydoc.info/gems/opentelemetry-sdk/OpenTelemetry/SDK/Trace/Span#finish-instance_method) used to be the last expression in this method and thus a span would normally be returned. Your change makes sense to me as this method should have a void return type.